### DISCUSSION TITLE
react-play app having errors

### DISCUSSION THREAD

**Discussion by mohitdevelops** [reactplay/react-play#574] [General]
Hi guys,
I've been struggling with creating my play for a couple of days, I've forked the repo and cloned it and installed all the dependencies, but while starting the app with 'npm start' the app is having tons of errors with a white screen, I've followed steps @Angryman18 suggested me but still, nothing is working out, now I have no idea what's going on here and how can I work on my play. 

Here are the errors I'm getting...
[localhost-1664361769488.log](https://github.com/reactplay/react-play/files/9671799/localhost-1664361769488.log)

**Comment by atapas** [comment]
Hi @mohitdevelops Could you please paste the screenshot of all the errors(not warnings) so that we start analyzing them?

**Reply by mohitdevelops** [reply]
Hi @atapas ,
Here are 3 SS of errors (not warnings), please have a look...

Error1
![error1](https://user-images.githubusercontent.com/51396626/192960619-b19e80f3-6ad0-4c94-8f73-b38d74d8e7c0.png)
Error2
![error2](https://user-images.githubusercontent.com/51396626/192960716-0cf1f36d-6d5e-4e55-8c10-1af421a2b5a0.png)
Error3
![error3](https://user-images.githubusercontent.com/51396626/192960751-5b0e9ce5-6374-4fde-b845-a68fab015b27.png)

**Reply by atapas** [reply]
Ok now tell me what's the react router version you have in the package.json file?

**Reply by atapas** [reply]
Also please go to the `node_modules` folder... find react-router dom there and see the version downloaded.

**Reply by mohitdevelops** [reply]
Its says: "react-router-dom": "6",

**Reply by atapas** [reply]
It should be "react-router-dom": "^6.4.0",

Thats how it is in main.. If it is not the case in your branch, then, your code is not correctly updated with the main. Please confirm.

**Reply by mohitdevelops** [reply]
There are 4 files in the react-router folder and in here the package.json file says this:--

{
  "_from": "react-router@6.4.1",
  "_id": "react-router@6.4.1",
  "_inBundle": false,
  "_integrity": "sha512-OJASKp5AykDWFewgWUim1vlLr7yfD4vO/h+bSgcP/ix8Md+LMHuAjovA74MQfsfhQJGGN1nHRhwS5qQQbbBt3A==",
  "_location": "/react-router",
  "_phantomChildren": {},
  "_requested": {
    "type": "version",
    "registry": true,
    "raw": "react-router@6.4.1",
    "name": "react-router",
    "escapedName": "react-router",
    "rawSpec": "6.4.1",
    "saveSpec": null,
    "fetchSpec": "6.4.1"
  },
  "_requiredBy": [
    "/react-router-dom"
  ],
  "_resolved": "https://registry.npmjs.org/react-router/-/react-router-6.4.1.tgz",
  "_shasum": "dd9cc4dfa264751d143a4b6c9d4faa60ab3ce26c",
  "_spec": "react-router@6.4.1",
  "_where": "D:\\my work\\React Projects\\react-play\\node_modules\\react-router-dom",
  "author": {
    "name": "Remix Software",
    "email": "hello@remix.run"
  },
  "bugs": {
    "url": "https://github.com/remix-run/react-router/issues"
  },
  "bundleDependencies": false,
  "dependencies": {
    "@remix-run/router": "1.0.1"
  },
  "deprecated": false,
  "description": "Declarative routing for React",
  "devDependencies": {
    "react": "^18.2.0"
  },
  "engines": {
    "node": ">=14"
  },
  "files": [
    "dist/",
    "CHANGELOG.md",
    "LICENSE.md",
    "README.md"
  ],
  "homepage": "https://github.com/remix-run/react-router#readme",
  "keywords": [
    "react",
    "router",
    "route",
    "routing",
    "history",
    "link"
  ],
  "license": "MIT",
  "main": "./dist/main.js",
  "module": "./dist/index.js",
  "name": "react-router",
  "peerDependencies": {
    "react": ">=16.8"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/remix-run/react-router.git",
    "directory": "packages/react-router"
  },
  "sideEffects": false,
  "types": "./dist/index.d.ts",
  "unpkg": "./dist/umd/react-router.production.min.js",
  "version": "6.4.1"
}

**Reply by atapas** [reply]
Ok but confirm the package.json of your branch says 6.4.1 or not

**Reply by mohitdevelops** [reply]
No it says 6 only, how can I update it to 6.4.1?
I just created my branch

**Reply by atapas** [reply]
Can you please paste your package.json content here or gimme your branch URL

**Reply by mohitdevelops** [reply]
Here is my package.json ----

{
  "name": "react-play",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@giscus/react": "^2.0.3",
    "@types/react": "^18.0.6",
    "@types/react-dom": "^18.0.2",
    "node-sass": "^7.0.1",
    "plop": "^3.0.5",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-helmet": "^6.1.0",
    "react-icons": "^4.3.1",
    "react-organizational-chart": "^2.1.1",
    "react-router-dom": "6",
    "react-scripts": "5.0.0",
    "react-shimmer-effect": "^1.0.9",
    "web-vitals": "^2.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "postbuild": "react-snap",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "create-play": "plop"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "reactSnap": {
    "puppeteerArgs": [
      "--no-sandbox",
      "--disable-setuid-sandbox"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "react-snap": "^1.23.0",
    "typescript": "^4.6.4"
  }
}

how to get branch url? I created and switched to it, see the SS -
![branch](https://user-images.githubusercontent.com/51396626/192965257-5768d901-5f23-43a7-b16e-3d62bb7381e5.png)

**Reply by atapas** [reply]
That's old and behind main. Please sync the code and do a fresh npm install. You must pull first from UPSTREAM

**Comment by Angryman18** [comment]
when did you fork? if you are using main your brach is 946 commits behind????? @mohitdevelops

![Screenshot from 2022-09-29 12-42-42](https://user-images.githubusercontent.com/63530626/192964418-40ea2957-240c-4ee4-bc96-edb80537551f.png)

**Reply by atapas** [reply]
Yes I am sensing that's the issue.. it's not updated with main.

**Reply by Angryman18** [reply]
yes i gues it is the issue. since its not synced.

**Reply by mohitdevelops** [reply]
I haven't committed anything yet, I started with new branch...

**Reply by Angryman18** [reply]
https://www.youtube.com/watch?v=VOakLctEC2Q

**Reply by atapas** [reply]
> I haven't committed anything yet, I started with new branch...

Doesn't matter.. Your repos main is behind upstream main. So first you need to fetch the changes and then to create new branch

**Reply by mohitdevelops** [reply]
Yes, now it's working, thank you so much, but now after hitting 'npm run start' I didn't see any prompt to 'create-react-play', It didn't ast to create a play. I've tried 'npm run create-react-play' also but nothing is happening + browse button going to not found.
